src/core/jpg.js

```diff
@@ -839,6 +862,13 @@ var JpegImage = (function JpegImageClosure() {
             break;
 
           case 0xFFDA: // SOS (Start of Scan)
+            numScanSegments++;
+            // A DNL marker (0xFFDC), if it exists, is only allowed at the end
+            // of the first scan segment and may only occur once in an image.
+            // Furthermore, to prevent an infinite loop, do *not* attempt to
+            // check DNL markers during re-parsing of the first scan segment.
+            const checkDNLMarker = numScanSegments === 1 && !dnlScanLines;
```

Comment:
`dnlScanLines !== null` to avoid infinite recursion when the value is `0`.

Reply:
I think you meant `dnlScanLines === null` here!?
Otherwise it will have the opposite effect, regarding the infinite recursion issue.

Given that zero isn't a valid value, which is now being enforced, I left this comparison intact.